packages/ra-core/src/form/FormDataConsumer.tsx

```diff
@@ -42,13 +41,20 @@ import warning from '../util/warning';
  *     </Edit>
  * );
  */
-const FormDataConsumer = (props: ConnectedProps) => {
-    const formData = useWatch();
-
-    return <FormDataConsumerView formData={formData} {...props} />;
+const FormDataConsumer = <TFieldValues extends FieldValues = FieldValues>(
+    props: ConnectedProps<TFieldValues>
+) => {
+    const formData = useWatch<TFieldValues>();
+    return (
+        <FormDataConsumerView<TFieldValues> formData={formData} {...props} />
```

Comment:
I have a TS warning here

![2022-12-08_10-31](https://user-images.githubusercontent.com/14542336/206410393-dafbdc87-cbf4-4c4e-bcb3-550266392222.png)

Reply:
Strange, because for me it didn't happen 🤔 😕 . But in rebase this was shown to me, and i fixed it. Can you try it again?
Fix commit -> https://github.com/marmelab/react-admin/pull/8389/commits/4c15b3793eac58c990db23bb227695960e736b60